bgpd/bgp_fsm.c

```diff
@@ -164,9 +164,7 @@ static struct peer *peer_xfer_conn(struct peer *from_peer)
 	 * on various buffers. Those need to be transferred or dropped,
 	 * otherwise we'll get spurious failures during session establishment.
 	 */
-	pthread_mutex_lock(&peer->io_mtx);
-	pthread_mutex_lock(&from_peer->io_mtx);
-	{
+	frr_with_mutex(&peer->io_mtx, &from_peer->io_mtx) {
```

Comment:
This works in LTR order right? And RTL for the unlock?

Reply:
Yeah, it's LTR / RTL ... but after some consideration I think adding this was maybe a bad idea & it'd be better to just use 2 `frr_with_mutex` lines.